src/simulators/matrix_product_state/matrix_product_state_internal.cpp

```diff
@@ -1705,13 +1705,39 @@ mps_container_t MPS::move_to_mps_container() {
                                        std::move(q_reg_[i].get_data(1))));
   }
   std::vector<std::vector<double>> lambda_vec;
-  for (auto i=0; i<num_qubits()-1; i++) {
+  for (uint_t i=0; i<num_qubits()-1; i++) {
     ret.second.push_back(std::move(lambda_reg_[i]));
   }
   initialize(MPS());
   return ret;
 }
 
+void MPS::initialize_from_mps(const mps_container_t &mps) {
+
+  uint_t num_qubits = mps.first.size();
+  // clear and restart all internal structures
+  q_reg_.clear();
+  lambda_reg_.clear();
+  q_reg_.resize(num_qubits);
+  lambda_reg_.resize(num_qubits-1);
+  qubit_ordering_.order_.clear();
+  qubit_ordering_.order_.resize(num_qubits);
+  std::iota(qubit_ordering_.order_.begin(), qubit_ordering_.order_.end(), 0);
+
+  qubit_ordering_.location_.clear();
+  qubit_ordering_.location_.resize(num_qubits);
+  std::iota(qubit_ordering_.location_.begin(), qubit_ordering_.location_.end(), 0);
+
+  // initialize values from mps_container_t
+  for (uint_t i=0; i<num_qubits; i++) {
+    MPS_Tensor next_tensor(mps.first[i].first, mps.first[i].second);
+    q_reg_[i] = next_tensor;
```

Comment:
If `data_t` in `MPS_Tensor` holds large `cmatrix_t`s, it is worth it to implement movement constructor and movement assignment operator for `MPS_Tensor`, to avoid extra copies of large matrices.

Reply:
@vvilpas , please review the changes I made in `initialize_from_mps`.